apps/src/lib/ui/SchoolInfoInterstitial.jsx

```diff
@@ -0,0 +1,206 @@
+        &&
+        (existingSchoolInfo.school_name || existingSchoolInfo.state || existingSchoolInfo.zip)
+      ) ? '-1' : '';
+
+    this.state = {
+      country: existingSchoolInfo.country || '',
+      schoolType: existingSchoolInfo.school_type || '',
+      schoolName: existingSchoolInfo.school_name || '',
+      schoolState: existingSchoolInfo.state || '',
+      schoolZip: existingSchoolInfo.zip || '',
+      schoolLocation: existingSchoolInfo.full_address || '',
+      ncesSchoolId: initialNcesSchoolId,
+      showSchoolInfoUnknownError: false,
+    };
+  }
+
+
+  handleSchoolInfoSubmit = () => {
+    let schoolData;
+    if (this.state.ncesSchoolId === '-1') {
+      schoolData = {
+        "user[school_info_attributes][school_name]": this.state.schoolName,
+        "user[school_info_attributes][school_state]": this.state.schoolState,
+        "user[school_info_attributes][school_zip]": this.state.schoolZip,
+        "user[school_info_attributes][full_address]": this.state.schoolLocation,
+      };
+    } else {
+      schoolData = {
+        "user[school_info_attributes][school_id]": this.state.ncesSchoolId,
+      };
+    }
+
+    const {formUrl, authTokenName, authTokenValue} = this.props.scriptData;
+    $.post({
+      url: formUrl + '.json',
+      dataType: "json",
+      data: {
+        '_method': 'patch',
+        [authTokenName]: authTokenValue,
+        "user[school_info_attributes][country]": this.state.country,
+        "user[school_info_attributes][school_type]": this.state.schoolType,
+        ...schoolData,
+      },
+    }).done(() => {
+      this.props.onClose();
+    }).fail(() => {
+      if (!this.state.showSchoolInfoUnknownError) {
+        // First failure, display error message and give the teacher a chance
+        // to try again.
+        this.setState({showSchoolInfoUnknownError: true});
+      } else {
+        // We already failed once, let's not block the teacher any longer.
+        // TODO (bbuchanan): Gather metrics on this.
+        this.props.onClose();
+      }
+    });
+  };
+
+  onCountryChange = (_, event) => {
+    const newCountry = event ? event.value : '';
+    this.setState({country: newCountry});
+  };
+
+  onSchoolTypeChange = (event) => {
+    const newType = event ? event.target.value : '';
+    this.setState({schoolType: newType});
+  };
+
+  onSchoolChange = (_, event) => {
+    const newSchool = event ? event.value : '';
+    this.setState({ncesSchoolId: newSchool});
+  };
+
+  onSchoolNotFoundChange = (field, event) => {
+    let newValue = event ? event.target.value : '';
+    this.setState({
+      [field]: newValue
+    });
+  };
+
+  render() {
+    return (
+      <BaseDialog
+        useUpdatedStyles
+        isOpen={true}
+        handleClose={this.props.onClose}
+        uncloseable
+      >
+        <div style={styles.container}>
+          <div style={styles.heading}>
+            We want to bring Computer Science to every student - help us track our progress!
```

Comment:
Does this need to be localized?

Reply:
Yes, thanks for the reminder. Added to the TODO list in the PR description.